tests/validator/validator.cpp

```diff
@@ -1981,26 +1981,60 @@ TEST(Validator, unitUserCreatedUnitsBananasAndApples)
     EXPECT_EQ_ISSUES(expectedIssues, validator);
 }
 
-TEST(Validator, unitIllDefinedEquivalentUnits)
+TEST(Validator, unitMissingEquivalentUnits)
 {
-    const std::vector<std::string> expectedIssues = {
+    const std::vector<std::string> expectedIssues1 = {
         "Variable 'v1' in component 'c1' does not have any units specified.",
         "Variable 'v2' in component 'c2' does not have any units specified.",
-        "Variable 'v1' in component 'c1' has units of '' and an equivalent variable 'v2' in component 'c2' with non-matching units of ''. The mismatch is: ",
     };
 
     libcellml::ValidatorPtr validator = libcellml::Validator::create();
     libcellml::ModelPtr m = createModelTwoComponentsWithOneVariableEach("m", "c1", "c2", "v1", "v2");
     auto c1 = m->component(0);
     auto c2 = m->component(1);
+
     auto v1 = c1->variable(0);
     auto v2 = c2->variable(0);
 
-    libcellml::Variable::addEquivalence(v1, v2);
+    m->addComponent(libcellml::Component::create("c3"));
+    auto v3 = libcellml::Variable::create("v3");
+    m->component("c3")->addVariable(v3);
+    v3->setUnits("second");
+    v3->setInterfaceType("public");
 
+    libcellml::Variable::addEquivalence(v1, v2);
+    libcellml::Variable::addEquivalence(v3, v1);
```

Comment:
No need for `v3` (and its owning component) anymore.

Reply:
541b8354